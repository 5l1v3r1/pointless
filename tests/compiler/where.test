
y = b where b = 2