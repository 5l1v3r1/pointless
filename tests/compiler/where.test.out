   0 [ SaveImports  ] 0       
   1 [ MakeEnv      ] 1       
   2 [ MakeNumber   ] 6       
   3 [ MakeDef      ] 0       (y)
   4 [ Jump         ] 12      (-> 16)
   5 [ MakeEnv      ] 1       
   6 [ MakeNumber   ] 8       
   7 [ MakeDef      ] 0       (b)
   8 [ Jump         ] 4       (-> 12)
   9 [ MakeNumber   ] 2       
  10 [ SaveVal      ] 0       (b)
  11 [ Return       ]
  12 [ LoadLocal    ] 0       (b)
  13 [ PopEnv       ]
  14 [ SaveVal      ] 0       (y)
  15 [ Return       ]
  16 [ MakeObj      ]
  17 [ FieldRef     ] 10      (output)
  18 [ Jump         ] 6       (-> 24)
  19 [ Dup          ]
  20 [ LangFieldRef ] 17      (!getHead)
  21 [ Pop          ] 1       
  22 [ Jump         ] 2       (-> 24)
  23 [ LangFieldRef ] 26      (!getTail)
  24 [ Dispatch     ]
